mlflow/models/evaluation/base.py

```diff
@@ -229,14 +257,7 @@ def data(self):
         if self._data is not None:
             return self._data
 
-        try:
-            from pyspark.sql import DataFrame as SparkDataFrame
-
-            spark_df_type = SparkDataFrame
-        except ImportError:
-            spark_df_type = None
-
-        if spark_df_type and isinstance(self._original_data, spark_df_type):
+        if self._spark_df_type and isinstance(self._original_data, self._spark_df_type):
             self._data = self._original_data.limit(
                 EvaluationDataset.SPARK_DATAFRAME_LIMIT
             ).toPandas()
```

Comment:
Have we verified how this behaves if the feature to evaluate is actually contained in the output from a VectorAssembler transformer stage? If the feature set is a single-column SparseVector or DenseVector, translating back to columns with a provided set of field names might be a useful quality of life feature. 
If this is completely out of scope, disregard this comment.

Reply:
Good point. 
We cannot support array-like feature value now. The Vector column in spark dataframe will be converted into a `Vector` instance in python side.

@dbczumar Shall we support a feature column which value is type of "ndarray" or list or "pyspark.ml.linalg.Vector" ?

Reply:
@WeichenXu123 Is there any additional challenge to supporting `ndarray` or `list`? For `pyspark.ml.linalg.Vector`, is the issue that `df.toPandas()` doesn't convert the vector to a list / ndarray?

Reply:
For information purposes:
```python
from pyspark.ml.feature import VectorAssembler

data = spark.table("ben_demo.wine_data")
assembler = VectorAssembler(inputCols=[col for col in data.columns if col != "class"], outputCol="features")
data_assembled = assembler.transform(data)
collected = data_assembled.toPandas()



type(collected['features'][0])

>> pyspark.ml.linalg.DenseVector
```
We'll have to do a bit of conversion here...
```python

data_convert_vector = data_assembled.withColumn("feature_array", vector_to_array("features"))
data_columns = [col for col in data_convert_vector.columns if col != "features"]
data_to_collect = data_convert_vector.select(*data_columns)
collected_data = data_to_collect.toPandas()


type(collected_data['feature_array'][0])

>> numpy.ndarray
```

Another tidbit: if we don't do the explicit casting or leave a DenseVector or SparseVector in the dataframe, Arrow will throw a warning that it can't process the vectorized representation and will default back to the much slower older version of collection of a Spark DataFrame to a Pandas DataFrame.

Reply:
Got it. For Spark, it sounds like we can document this shortcoming for now and let it fail. For Pandas, I'm still not sure I see the challenge with supporting `ndarray` and `list` columns.

Reply:
ndarray/list/pyspark.ml.Vector as element type all supported now.